src/store/index.ts

```diff
@@ -80,41 +90,103 @@ export default createStore<State>({
       const res = await axios.post<UploadResponse>(
         "https://rtc-data.registree.io/data/uploads",
         request
       );
-      if (res.status !== 200) {
+      const enclavePubKey = getters.enclavePublicKey;
+      const ourSecretKey = state.ourSecretKey;
+
+      if (res.status !== 200 || !enclavePubKey || !ourSecretKey) {
+        // TODO: Error Handling
+        return "error";
+      }
+
+      const { nonce, ciphertext } = res.data;
+
+      const msg = decryptMessage(
+        ciphertext,
+        nonce,
+        enclavePubKey,
+        ourSecretKey
+      );
+
+      if (!msg) {
+        // TODO: Error Handling
         return "error";
       }
-      await dispatch("decryptUploadResponse", res.data);
+
+      await dispatch("parseUploadMessage", msg);
     },
-    async decryptUploadResponse(
-      { state, dispatch, getters },
-      response: UploadResponse
-    ) {
+    async postAccessForm({ commit, getters, state }, request: UploadRequest) {
+      const res = await axios.post<UploadResponse>(
+        "https://rtc-data.registree.io/auth/tokens",
+        request
+      );
       const enclavePubKey = getters.enclavePublicKey;
-      if (!enclavePubKey || !state.ourSecretKey) {
+      const ourSecretKey = state.ourSecretKey;
+
+      if (res.status !== 200 || !enclavePubKey || !ourSecretKey) {
+        // TODO: Error Handling
         return "error";
       }
+
+      const { nonce, ciphertext } = res.data;
+
       const msg = decryptMessage(
-        response.ciphertext,
-        response.nonce,
+        ciphertext,
+        nonce,
         enclavePubKey,
-        state.ourSecretKey
+        ourSecretKey
       );
+
       if (!msg) {
         // TODO: Error Handling
         return "error";
       }
-      await dispatch("parseUploadMessage", msg);
+
+      //TODO: Open Dialog Box show message
+
+      commit("setDecryptedMsg", msg);
+    },
+    async decryptResponse({ state, getters }, response: UploadResponse) {
+      const enclavePubKey = getters.enclavePublicKey;
+      if (!enclavePubKey || !state.ourSecretKey) {
+        // TODO: Error Handling
+        return "error";
+      }
+      return decryptMessage(
+        response.ciphertext,
+        response.nonce,
+        enclavePubKey,
+        state.ourSecretKey
+      );
     },
     async parseUploadMessage({ commit }, message: Uint8Array) {
       console.log("parseUploadMessage:", message);
       const accessKey = btoa(message.slice(0, 24).toString());
       const uuid = message.slice(24);
       commit("saveUploadResult", { accessKey, uuid });
+    },
+    async requestExecutionToken({ commit, dispatch, getters }, data) {
+      const enclavePubKey = getters.enclavePublicKey;
+      if (!enclavePubKey) {
+        return null;
+      }
+      const { ourData, messageData } = await encryptJson(
+        data,
+        enclavePubKey as Base64
+      );
+
+      commit("saveSecretKey", ourData.ourSecretKey);
+      dispatch("postAccessForm", {
```

Comment:
```suggestion
      await dispatch("postAccessForm", {
```

(The "loading" state of the execution token request / submit button awaits this.)